kbase-extension/static/kbase/js/common/cellComponents/paramsWidget.js

```diff
@@ -0,0 +1,551 @@
+                                    }
+                                });
+                            } else {
+                                return null;
+                            }
+                        }
+                    });
+
+                    fieldWidget.bus.respond({
+                        key: {
+                            type: 'get-parameters'
+                        },
+                        handle: (message) => {
+                            if (message.parameterNames) {
+                                return Promise.all(
+                                    message.parameterNames.map((paramName) => {
+                                        return paramsBus.request({
+                                            parameterName: paramName
+                                        }, {
+                                            key: {
+                                                type: 'get-parameter'
+                                            }
+                                        })
+                                            .then((result) => {
+                                                let returnVal = {};
+                                                returnVal[paramName] = result.value;
+                                                return returnVal;
+                                            });
+                                    })
+                                )
+                                    .then((results) => {
+                                        let combined = {};
+                                        results.forEach((res) => {
+                                            Object.keys(res).forEach((key) => {
+                                                combined[key] = res[key];
+                                            });
+                                        });
+                                        return combined;
+                                    });
+                            }
+                        }
+                    });
+
+                    // Just pass the update along to the input widget.
+                    paramsBus.listen({
+                        key: {
+                            type: 'update',
+                            parameter: parameterSpec.id
+                        },
+                        handle: function (message) {
+                            fieldWidget.bus.emit('update', {
+                                value: message.value
+                            });
+                        }
+                    });
+
+                    return fieldWidget;
+                });
+        }
+
+        function renderAdvanced() {
+            const areaElement = 'parameters-area',
+                areaSelector = '[data-element="' + areaElement + '"]',
+                advancedInputs = container.querySelectorAll(areaSelector + ' [data-advanced-parameter]');
```

Comment:
Where is `data-advanced-parameter` added to inputs?

Reply:
i'm not sure to be honest, i know that the app spec has fields which mark it as advanced, but i can't find in the code where the element is actually added. there's some magic that happens in the conversion of the app spec within that common/spec and common/props methods. @briehl may know better